FWCore/Framework/src/Principal.cc

```diff
@@ -295,6 +294,24 @@ namespace edm {
     }
   }
 
+  Principal::Principal(std::shared_ptr<ProductRegistry const> reg,
+                       std::vector<std::shared_ptr<ProductResolverBase>> const& resolvers,
+                       ProcessConfiguration const& pc,
+                       BranchType bt,
+                       HistoryAppender* historyAppender)
+      : EDProductGetter(),
+        processHistoryPtr_(),
+        processHistoryID_(),
+        processHistoryIDBeforeConfig_(),
+        processConfiguration_(&pc),
+        productResolvers_(resolvers.begin(), resolvers.end()),
```

Comment:
Note, the type of `productResolvers_` and `resolvers` is slightly different (since `productResolvers_` uses the `edm:propagate_const` on the elements) which is why I need to use the iterators.